MIGRATION_v2-to-v3.md

```diff
@@ -0,0 +1,152 @@
+# Migration Guide for v2 to v3
+
+If you are using v2 and want to upgrade to v3, we have tried to keep the changes minimum. We did this by keeping the prop shapes of smart-components in the new modules. We tried to simplify certain aspects by removing certain arguments from renderProps(especially in renders that deal with messasge, input and title), but it should be easily migrated if you follow this guide
+
+## Common changes
+* Installation `npm i @sendbird/uikit-react`
+* There are no changes to SendbirdProvider, App, withSendbird, useSendbirdStateContext etc,
+* import path should start from `"@sendbird/uikit-react`!
+
+## ChannelList
+
+Can be imported as `import { ChannelList } from "@sendbird/uikit-react"`
+or `import ChannelList from "@sendbird/uikit-react/ChannelList"`
+
+### Changes - none
+### Added props
+  * renderPlaceHolderError?: (props: void) => React.ReactNode;
+  * renderPlaceHolderLoading?: (props: void) => React.ReactNode;
+  * renderPlaceHolderEmptyList?: (props: void) => React.ReactNode;
+
+## Channel
+
+Can be imported as `import { Channel } from "@sendbird/uikit-react"`
+or `import Channel from "@sendbird/uikit-react/Channel"`
+
+### Changes
+  * Removed renderCustomMessage, renderChatItem. Use renderMessage instead
+  Current channel can be obtained from `useChannel()`
+  Message related actions should be implemented using sendbirdSelectors
+  ```jsx
+    <Channel renderMessage={MyAdminMessageComponent} />
+    const MyAdminMessageComponent = ({message, chainTop, chainBottom }) => {
+      const {
+        currentGroupChannel,
+        scrollToMessage,
+      } = useChannel();
+      const globalStore = useSendbirdStateContext();
+      // use sendbirdSelectors and globalStore to implement onDeleteMessage, onUpdateMessage, onResendMessage
+      const updateMessage = sendBirdSelectors.getDeleteMessage(globalStore);
+      if (message.messageType === 'admin') {
+        return (
+          ... implement your component
+        );
+      }
+      return null;
+    }
+  ```
+  * Changed signature of renderMessageInput to `() => React.ReactNode`
+    Current channel can be obtained from `useChannel()`
+    Message related actions should be implemented using sendbirdSelectors
+  ```jsx
+    <Channel renderMessageInput={MyMessageInput} />
+    const MyMessageInput = ({message, chainTop, chainBottom }) => {
+      const {
+        currentGroupChannel,
+      } = useChannel();
+      const globalStore = useSendbirdStateContext();
+      // use sendbirdSelectors and globalStore to implement onDeleteMessage, onUpdateMessage, onResendMessage
```

Comment:
Examples need to be changed
How about metioning `getSendUserMessage` and `getSendFileMessage`?